asv/config.py

```diff
@@ -48,13 +48,16 @@ def __init__(self):
     def load(cls, path=None):
         """
         Load a configuration from a file.  If no file is provided,
-        defaults to `asv.conf.json`.
+        defaults to `asv.conf.json` (also checks for `asv.conf.jsonc`).
         """
         if not path:
             path = "asv.conf.json"
 
         if not os.path.isfile(path):
-            raise util.UserError(f"Config file {path} not found.")
+            # also check for `asv.conf.jsonc` (JSON with comments)
+            path_c = path + 'c'
+            if not os.path.isfile(path_c):
+                raise util.UserError(f"Config file {path} not found.")
```

Comment:
I guess my main concern with this is that it'll silently ignore the `.jsonc` file if there's a `.json` with comments as well. Could you refactor this a bit more? Maybe a separate helper function which has a list of Path().extensions to check and raises if more than one is found.